DataProcessing/ExtractAlphaTrueBeatsConverter.cs

```diff
@@ -0,0 +1,460 @@
+
+            do
+            {
+                // Once we begin processing "FQ1", this variable will be equal
+                // to false and end the loop.
+                allQuarters = !allQuarters;
+
+                foreach (var earningsMetric in _earningsMetrics)
+                {
+                    foreach (var line in GetTrueBeatsRawLines(earningsMetric, processingDate, allQuarters))
+                    {
+                        if (!char.IsNumber(line.FirstOrDefault()))
+                        {
+                            // Skips the CSV header line and empty lines
+                            continue;
+                        }
+
+                        var csv = line.Split(',');
+
+                        var ticker = csv[_tickerIndex];
+                        if (!trueBeats.TryGetValue(ticker, out var symbolTrueBeats))
+                        {
+                            // Not all tickers appear in the fiscal periods dataset, so we'll need to initialize
+                            // a new list for these tickers that won't have fiscal year information.
+                            symbolTrueBeats = new List<ExtractAlphaTrueBeat>();
+                            trueBeats[ticker] = symbolTrueBeats;
+                        }
+                        if (!outputData.TryGetValue(ticker, out var outputTrueBeats))
+                        {
+                            outputTrueBeats = new List<ExtractAlphaTrueBeat>();
+                            outputData[ticker] = outputTrueBeats;
+                        }
+
+                        var (fiscalYear, fiscalQuarter) = ParseFiscalPeriod(csv[_fiscalPeriodIndex]);
+                        if (!allQuarters && symbolTrueBeats.Any(x =>
+                            x.Time == processingDate &&
+                            x.EarningsMetric == earningsMetric &&
+                            x.FiscalPeriod.FiscalYear == fiscalYear &&
+                            x.FiscalPeriod.FiscalQuarter == fiscalQuarter &&
+                            x.ExpertBeat != null &&
+                            x.TrendBeat != null &&
+                            x.ManagementBeat != null))
+                        {
+                            // We've encountered a duplicate Symbol while processing the FQ1 TrueBeats dataset, and
+                            // the first piece of data has already been included.
+                            // Majority of the duplicated symbols in the raw data will have duplicate data
+                            // that equals another line in the same file. However, sometimes this isn't true
+                            // and the TrueBeat calculation fluctuates between two numbers.
+                            if (_duplicateFQ1DataTickers.Add(ticker))
+                            {
+                                Log.Error($"ExtractAlphaTrueBeatsConverter.ParseTrueBeats(): Duplicate data encountered in FQ1 dataset for ticker: {ticker} - skipping");
```

Comment:
Could these be updates emitted at different times? Makes me wonder if the Time of the baseData being used is the right one

Reply:
I don't think so, there's no way to know which event came first, and the fact that data is duplicated on multiple rows is suspicious since almost every single ticker in that data doesn't have duplicates